### PR TITLE
feat: applepay payment request object

### PR DESCRIPTION
**Pull request by Sangamesh26**
## Type of Change

- [x] New feature

## Description

Return applepay request object in the session response.

## Motivation and Context

The payment request object should be produced from the backend as the frontend does not have access to it.

## How did you test it?

Manual
Screenshot 2023-02-07 at 8 23 02 PM
Screenshot 2023-02-07 at 8 23 49 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

File: crates/api_models/src/payments.rs [hunk 1]
```diff
@@ -1120,31 +1120,64 @@ pub enum SessionToken {
...
-        /// A string which represents the properties of a payment
-        signature: String,
-        /// The identifier for the operational analytics
-        operational_analytics_identifier: String,
-        /// The number of retries to get the session response
-        retries: u8,
-        /// The identifier for the connector transaction
-        psp_id: String,
+        /// Session object for Apple Pay
+        session_object: Box<ApplePaySessionObject>,
```
Comment (jarnura) [inline crates/api_models/src/payments.rs:1124]: Boxed all the field in each variant of the enum, or create object and for each variant and box them.

File: crates/router/src/connector/applepay/transformers.rs [hunk 2]
```diff
@@ -55,48 +87,133 @@ impl TryFrom<&types::PaymentsSessionRouterData> for ApplepaySessionRequest {
...
+            .to_owned()
+            .get_required_value("connector_meta_data")
+            .change_context(errors::ConnectorError::NoConnectorMetaData)?;
+
+        let metadata: ApplePayMetaData = metadata
+            .parse_value("ApplePayMetaData")
+            .change_context(errors::ConnectorError::RequestEncodingFailed)?;
+
+        let amount_info = AmountInfo {
+            label: "Demo".to_string(),
```
Comment (jarnura) [inline crates/router/src/connector/applepay/transformers.rs:134]: Why these where hard coded?

File: crates/router/src/connector/applepay/transformers.rs [hunk 3]
```diff
@@ -55,48 +87,133 @@ impl TryFrom<&types::PaymentsSessionRouterData> for ApplepaySessionRequest {
...
+            amount: (item.data.request.amount / 100).to_string(),
+        };
+
+        let payment_request = PaymentRequest {
+            country_code: item
+                .data
+                .request
+                .country
+                .to_owned()
+                .unwrap_or_else(|| "US".to_string()),
```
Comment (jarnura) [inline crates/router/src/connector/applepay/transformers.rs:145]: By the spec `US` is default country or by the connector documentation `US` is default country?

**Review by jarnura — CHANGES_REQUESTED** [review]

File: crates/api_models/src/payments.rs (hunk 1 above)
Comment (Sangamesh26) [reply crates/api_models/src/payments.rs]: Done

File: crates/router/src/connector/applepay/transformers.rs (hunk 2 above)
Comment (Sangamesh26) [reply crates/router/src/connector/applepay/transformers.rs]: label type is always final, label hardcoding is removed as it appears on their bank or credit card statement.

File: crates/router/src/connector/applepay/transformers.rs (hunk 3 above)
Comment (jarnura) [reply crates/router/src/connector/applepay/transformers.rs]: Address the above comment.

Comment (Sangamesh26) [reply crates/router/src/connector/applepay/transformers.rs]: I wasn't sure about the country, so added US as default, now it is changed to getting even country from the merchant.

**Review by jarnura — APPROVED** [review]

**Review by SanchithHegde — DISMISSED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by ashokkjag — APPROVED** [review]

**Pull request merged**